### PR TITLE
ACCUMULO-4630 Moving Accumulo documentation to website

### PR DESCRIPTION
**Pull request by mikewalch**

### CODE REVIEW

**Comment by milleruntime** [comment]
+1 Mike gave me a quick demo and it looks good.

**Comment by keith-turner** [comment]
What chnages will we make to the website when 2.0.0 is released?

**Comment by mikewalch** [comment]
@keith-turner, we'll need to make the following changes:

```bash
cp -r _docs-unreleased _docs-2.0  # create 2.0 copy of documentation
cp _layouts/doc-unreleased.html _layouts/doc-2.0.html # create 2.0 document layout
vim _config.yml # configure 2.0 (like unreleased documentation)
```

**Comment by mikewalch** [comment]
I pushed a commit that changes where unreleased docs can be viewed.  

From now on, use the URL below:

[http://0.0.0.0:4000/docs/unreleased/](http://0.0.0.0:4000/docs/unreleased/)

File: _docs-unreleased/development/iterator_design.md [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+The common failure condition is referred to as a "re-seek". In the context of a Scan, TabletServers construct the
+"stack" of Iterators and batch up Key-Value pairs to send back to the client. When a sufficient number of Key-Value
+pairs are collected, it is common for the Iterators to be "torn down" until the client asks for the next batch of
+Key-Value pairs. This is done by the TabletServer to add fairness in ensuring one Scan does not monopolize the available
+resources. When the client asks for the next batch, the implementation modifies the original Range so that servers know
+the point to resume the iteration (to avoid returning duplicate Key-Value pairs). Specifically, the new Range is created
+from the original but is shortened by setting the startKey of the original Range to the Key last returned by the Scan,
+non-inclusive.
+
+### `seek`'ing backwards
```
Comment (keith-turner) [inline _docs-unreleased/development/iterator_design.md]: The way seek is formatted in this header seems odd

**Comment by keith-turner** [comment]
I just compared this change side by side with 1.8 user manual.  **Its an enormous improvement.**  I really like the collapsible menus on the left (in the previous docs, it was just a giant expanded multi-level list).  I also like how you reorganized the content.  I think some minor adjustments can be made in the future, but this is a huge step in the right direction.  Also, the changes will make is easier to reorg the content incrementally.

**Pull request merged**